### ISSUE TITLE
disabling parseTransFromString

### ISSUE THREAD

**Issue by hobobobo** [i18next/i18next-scanner#133]
#### Version
- i18next: <i>15.0.4</i>
- i18next-scanner: <i>2.9.1</i>

#### Configuration
```js
{ // i18next-scanner options
options: {
    debug: true,
    func: {
      list: ['i18n.__'],
      extensions: ['.js'],
    },
    trans: false,
    lngs: ['en', 'de'],
    ns: [
      'scanned',
    ],
    defaultNs: 'scanned',
    defaultValue: '__STRING_NOT_TRANSLATED__',
    resource: {
      loadPath: 'src/locales/{{lng}}.json',
      savePath: 'src/locales/{{ns}}_{{lng}}.json',
      jsonIndent: 2,

    },
    nsSeparator: false, // namespace separator
    keySeparator: false, // key separator
    interpolation: {
      prefix: '{{',
      suffix: '}}',
    },
  },
}
```
In documentation we can see
trans 
Type: Object or false
You can set trans to false to disable parsing Trans component as below:
```js
{
    trans: false
}
```
But it NOT works. 

The parser will be disabled if extensions are empty
```js
trans {
   extensions: []
}
```

**Comment by tkforce** [comment]
Same issue here.

I think the reason is when merging the default options with given options in `transformOptions(options)` , the code will set the `trans.xxx` to default value if trans.xxx is undefined.
```
if (_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans.component'))) {
    _lodash2.default.set(options, 'trans.component', defaults.trans.component);
}
if (_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans.i18nKey'))) {
    _lodash2.default.set(options, 'trans.i18nKey', defaults.trans.i18nKey);
}
if (_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans.defaultsKey'))) {
    _lodash2.default.set(options, 'trans.defaultsKey', defaults.trans.defaultsKey);
}
if (_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans.extensions'))) {
    _lodash2.default.set(options, 'trans.extensions', defaults.trans.extensions);
}
if (_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans.fallbackKey'))) {
    _lodash2.default.set(options, 'trans.fallbackKey', defaults.trans.fallbackKey);
}
```
Naturally, when we try to turn off the <Trans> feature with  `trans: false`, the `transformOptions(options)` will fill the options under `trans` with default options.

Maybe simply adding an if statement will be suffice?
```
if(_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans'))){
    if (_lodash2.default.isUndefined(_lodash2.default.get(options, 'trans.component'))) {
        _lodash2.default.set(options, 'trans.component', defaults.trans.component);
    }
  ...
}
```

**Comment by cheton** [comment]
Published v2.9.2. Thank you!
